applications/FluidDynamicsApplication/python_scripts/apply_wall_law_process.py

```diff
@@ -0,0 +1,152 @@
+        default_settings = KratosMultiphysics.Parameters("""{
+            "model_part_name" : "",
+            "wall_model_name" : "",
+            "wall_model_settings" : {},
+            "calculate_normals_at_each_step" : false
+        }""")
+        return default_settings
+
+    def ExecuteInitialize(self):
+        # Get data from settings
+        model_part =  self.model[self.settings["model_part_name"].GetString()]
+        wall_model_name = self.settings["wall_model_name"].GetString()
+
+        # Get the condition registry name
+        # Note that in here we are assuming a unique condition geometry in model part
+        pts_num = 0
+        for condition in model_part.Conditions:
+            pts_num = condition.GetGeometry().PointsNumber()
+            break
+        pts_num = model_part.GetCommunicator().GetDataCommunicator().MaxAll(pts_num)
+        domain_size = model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        cond_aux_name = self.__wall_model_condition_name_map[wall_model_name]
+        cond_reg_name = f"{cond_aux_name}{domain_size}D{pts_num}N"
+
+        # Substitute current conditions by the corresponding ones implementing the wall model
+        # 1. Save current condition data in a map such that Id : (geometry, properties)
+        aux_cond_data_map = {}
+        for condition in model_part.Conditions:
+            condition.Set(KratosMultiphysics.TO_ERASE,True)
+            aux_ids = [node.Id for node in condition.GetNodes()]
+            aux_props = condition.Properties
+            aux_cond_data_map[condition.Id] = (aux_ids, aux_props)
+        # 2. Remove current conditions
+        # Note that in here we need to assume that no submodelpart is hanging from this one as the hierarchical structure would be destroyed
+        if model_part.NumberOfSubModelParts() != 0:
+            raise Exception(f"Wall model part '{model_part.FullName}' has submodelparts.")
+        model_part.RemoveConditionsFromAllLevels(KratosMultiphysics.TO_ERASE)
+        # 3. Create new wall law conditions from the data map info
+        for key, value in aux_cond_data_map.items():
+            model_part.CreateNewCondition(cond_reg_name, key, value[0], value[1])
+
+        # Set the SLIP flag in the wall model part nodes and calculate the nodal normals
+        # This is required in order to impose the no penetration condition by the rotating schemes
+        KratosMultiphysics.NormalCalculationUtils().CalculateOnSimplex(model_part, domain_size) #FIXME: This may interact with the nodal normals of slip boundaries (e.g. floor-wall in a 3D channel)
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.SLIP, True, model_part.Nodes)
+
+        # Set the WALL flag in the wall model part new conditions
+        # This is required in order to add the wall model contribution within the condition implementation
+        KratosMultiphysics.VariableUtils().SetFlag(KratosMultiphysics.WALL, True, model_part.Conditions)
```

Comment:
Nice :D

Reply:
What if we set the Flag of the geometry as well in here? It would be useful in some cases...

Reply:
I don't think this is a good idea as the same geometry can be used to create any number of conditions, which might implement a complete different physics. Think for instance in the volumetric heat + fluid coupling.